### PR TITLE
refacto(log): #CO-1699 add better logs

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Added more logs and modified what already exists to be more detailed.

## Issue ticket number and link
[CO-1699](https://jira.support-ent.fr/browse/CO-1699)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/supportpivot/constants/BusConstants.java [hunk 1]
```diff
@@ -1,7 +0,0 @@
-package fr.openent.supportpivot.constants;
-
-public class BusConstants {
-
-    public static final String OK = "ok";
-    public static final String STATUS = "status";
-}
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/constants/BusConstants.java:7]: Unused variable

File: src/main/java/fr/openent/supportpivot/constants/ConfigField.java [hunk 2]
```diff
@@ -13,7 +13,6 @@ public class ConfigField {
     public static final String STATUS_EXTERNAL = "status_external";
     public static final String RESOLUTION_EXTERNAL = "resolution_external";
     public static final String JIRA_STATUS_MAPPING = "jira-status-mapping";
-    public static final String STATUTSJIRA = "statutsJira";
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/constants/ConfigField.java:16]: Unused variable

File: src/main/java/fr/openent/supportpivot/constants/JiraConstants.java [hunk 3]
```diff
@@ -28,7 +28,4 @@ public class JiraConstants {
 
     public static final String REPORTER_ENT = "assistanceMLN";
     public static final String REPORTER_LDE = "lde";
-
-    public static final String JIRA_STATUS_MAPPING = "jira-status-mapping";
-    public static final String STATUTS_DEFAULT = "statutsDefault";
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/constants/JiraConstants.java:33]: Unused variable

File: src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultDemandeServiceImpl.java [hunk 4]
```diff
@@ -342,7 +347,7 @@ private void sendToJIRA(final HttpServerRequest request, JsonObject jsonPivotIn,
 
                     //return to IWS ticket with Id_JIRA
                     sendToIWS(request,
-                            eitherJsonPivotOut.right().getValue().getJsonObject("jsonPivotCompleted"),
+                            eitherJsonPivotOut.right().getValue().getJsonObject(Field.JSONPIVOTCOMPLETED , new JsonObject()),
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultDemandeServiceImpl.java:350]: Utilisation de Field et ajout valeur par defaut

File: src/main/java/fr/openent/supportpivot/managers/ServiceManager.java [hunk 5]
```diff
@@ -47,6 +47,7 @@ private ServiceManager(Vertx vertx, JsonObject config) {
                 routeurService = new CrifRouterService(httpClientService,jiraService);
                 break;
             default:
+                //todo delete
                 log.error("Unknown value when starting Pivot Service. collectivity: " + ConfigManager.getInstance().getConfig().getCollectivity());
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/managers/ServiceManager.java:51]: IWS sera supprimer dans les prochains ticket

File: src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultJiraServiceImpl.java [hunk 6]
```diff
@@ -167,12 +168,11 @@ private void terminateRequest(HttpClientRequest httpClientRequest) {
      * @param jsonPivotIn JSON in pivot format
      * @param handler     return JsonPivot from Jira or error message
      */
+    @Deprecated
     public void sendToJIRA(final JsonObject jsonPivotIn, final Handler<Either<String, JsonObject>> handler) {
-
         //ID_IWS is mandatory
         if (!jsonPivotIn.containsKey(IDIWS_FIELD)
                 || jsonPivotIn.getString(IDIWS_FIELD).isEmpty()) {
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultJiraServiceImpl.java:175]: IWS sera supprimer dans les prochains ticket

File: src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultJiraServiceImpl.java [hunk 7]
```diff
@@ -449,13 +448,16 @@ public void updateJiraTicket(final JsonObject jsonPivotIn, final String jiraTick
...
                                     return;
                                 }
 
                                 //Is JIRA ticket had been created by same IWS issue ?
                                 String jsonPivotIdIWS = jsonPivotIn.getString(IDIWS_FIELD);
                                 if (!jiraTicketIdIWS.equals(jsonPivotIdIWS)) {
+                                    //todo delete
                                     handler.handle(new Either.Left<>("102;JIRA Ticket " + jiraTicketId + " already link with an another IWS issue"));
                                     return;
                                 }
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/deprecatedservices/DefaultJiraServiceImpl.java:463]: IWS sera supprimer dans les prochains ticket

File: src/main/java/fr/openent/supportpivot/model/status/JiraStatus.java [hunk 8]
```diff
@@ -3,13 +3,17 @@
...
 public class JiraStatus extends StatusModel {
     private final List<String> nameList;
 
     public JiraStatus(String key, JsonArray nameList) {
         super(key);
-        this.nameList = nameList.getList();
+        this.nameList = nameList.stream()
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .collect(Collectors.toList());
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/status/JiraStatus.java:16]: Supprime les éléments du JsonArray qui ne sont pas des Strings.

File: src/main/java/fr/openent/supportpivot/services/MongoService.java [hunk 9]
```diff
@@ -28,22 +29,22 @@ public MongoService(String mongoCollection) {
...
         jsonPivot.put(Field.SOURCE, source);
         jsonPivot.put(Field.DATE, dateFormat.format(date));
 
-        mongo.insert(mongoCollection, jsonPivot, retourJson -> {
-            if (!BusConstants.OK.equals(retourJson.body().getString(BusConstants.STATUS))) {
-                log.error("Supportpivot : could not save json to mongoDB");
+        mongo.insert(mongoCollection, jsonPivot, MongoDbResult.validActionResultHandler(event -> {
+            if (event.isLeft()) {
+                String message = EitherHelper.getOrNullLeftMessage(event);
+                log.error(String.format("[SupportPivot@%s::saveTicket] Could not save json to mongoDB %s", this.getClass().getName(), message));
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/MongoService.java:42]: Si le mongo.insert remonte une erreur elle ne sera pas remonter, a voir pour remonter l'erreur dans les futures tickets

File: src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java [hunk 10]
```diff
@@ -276,8 +285,10 @@ public JsonObject PrepareTicketForCreation(JsonObject jsonPivotIn) {
             title = jsonPivotIn.getString(TITLE_FIELD);
         }
 
-        List<String> newModules = ((List<String>) jsonPivotIn.getJsonArray(MODULES_FIELD, new JsonArray()).getList()).stream()
-                .map(module -> moduleEntToPivot(module))
+        List<String> newModules = jsonPivotIn.getJsonArray(MODULES_FIELD, new JsonArray()).stream()
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .map(this::moduleEntToPivot)
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java:291]: Supprime les éléments du JsonArray qui ne sont pas des Strings.

**Review by JinDouang — APPROVED** [review]

**Pull request merged**